src/LBPair.sol

```diff
@@ -900,24 +923,39 @@ contract LBPair is LBToken, ReentrancyGuardUpgradeable, Clone, ILBPair {
      * @dev Returns the address of the token X
      * @return The address of the token X
      */
-    function _tokenX() internal pure returns (IERC20) {
-        return IERC20(_getArgAddress(0));
+    function _tokenX() internal view returns (IERC20) {
+        address tokenX_;
+        bytes32 slot = 0x3441ab29b24daf7a3fd59500b0e08396ec08ec96f5cc2d0362924cdd45cfec31; //keccak256(abi.encode(uint256(keccak256("sovrynlbdex.pair.storage.TokenX")) - 1));
```

Comment:
Can this not be a constant?

Reply:
agree - better move to a constant this and all other unstructured storage slots

Reply:
Updated in this [commit](https://github.com/DistributedCollective/sovryn-lb-dex/pull/3/commits/ab08ddafa24bc76b0d41843f366ac1d6fcf439ff)